flocker/provision/_digitalocean.py

```diff
@@ -0,0 +1,373 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+DigitalOcean provisioner.
+"""
+import time
+from functools import partial
+
+from ._libcloud import LibcloudProvisioner
+from ._install import (
+    provision, run,
+    task_install_digitalocean_kernel, DIGITALOCEAN_KERNEL
+)
+from ._common import Kernel
+
+
+def retry_on_error(error_checkers, callable, *args, **kwargs):
```

Comment:
That is, it should fail after some number of iterations, or some amount of time.